### DISCUSSION TITLE
gokrazy with CasaOS

### DISCUSSION THREAD

**Discussion by m-abdalrahman** [IceWhaleTech/CasaOS#281] [Ideas]
by using [gokrazy](https://github.com/gokrazy/gokrazy), we can deploy CasaOS as appliances to a Raspberry Pi or PC, and zimaboard 'maybe in the future'.
I think this will produce a complete O.S

**Comment by Protektor-Desura** [comment]
Umm no it only handles go programs/code. Also Zimaboards are X86_64 so they would be supported no problem. But you couldn't use gokrazy by itself to make the entire OS. For example the Linux kernel isn't written in GO, so you can not use gokrazy to compile the Linux kernel. There are large numbers of other code parts of the OS that aren't written in Go as well.

**Reply by m-abdalrahman** [reply]
what I mean is that it is possible to use gokarzy to deploy CasaOS as appliances and this will imagine us we are working on a complete OS based on the idea of ​​CasaOS that relies on docker images

**Reply by Protektor-Desura** [reply]
The idea as I understand it is eventually an ISO or install image you can get a full OS going with the majority of the apps being installed via docker rather than DEB files or whatever installed directly on the Host OS. The full OS would basically be just enough software to get docker running and maybe a few other tools, but the majority of things would run in docker.

**Reply by m-abdalrahman** [reply]
yes, that what I mean, and I think at this time this is the Idea for CasaOS

**Reply by m-abdalrahman** [reply]
and instead of trying to run CasaOS in different distribution and test it if working, now we have CasaOS on ISO file with running CasaOS
